src/julia/generic.jl

```diff
@@ -181,6 +181,21 @@ Base.convert(::Type{Array}, xd::DistMatrix{T}) where {T} =
 
 Base.Array(xd::DistMatrix) = convert(Array, xd)
 
+# super slow, but handy?
+function Base.setindex!(A::DistMatrix,
+                        values,
+                        globalis,
+                        globaljs)
+    if typeof(values) <: Number
```

Comment:
Maybe better to split out to a separate method. Alternatively, make it `values isa Number`. Might also be useful to take a look at how this is handled for CuArrays where there is a similar concern.